tools/awps-tunnel/client/src/panels/RequestHistory.tsx

```diff
@@ -126,28 +128,93 @@ export function RequestHistory(props: RequestHistoryProps) {
   );
 }
 
+const renderContent = (message: {headers: Record<string, string>, content: string}): ReactNode => {
+  if(message.headers['Content-Type'] === 'application/json' || message.headers['Content-Type'] === 'text/json'){
+    try {
+      const parsedJson = JSON.parse(message.content);
+      return <div><ReactJson src={parsedJson} collapsed={1}/></div>;
+    } catch (error) {
+    }
+  }
+  return <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+    {message.content}
+  </div>
+};
+
+function revertRawTextToMessage(rawText: string):{ headers: Record<string, string>, content: string }{
+  const lines: string[] = rawText.split('\n').map(line => line.trim());
+  const emptyIndex: number = lines.indexOf('');  // 找到空行，它分隔了headers和body
+  let headers: Record<string, string> = {};
+  let content: string = '';
+  for (let i: number = 0; i < emptyIndex; i++) {
+    const splitIndex: number = lines[i].indexOf(':');
+    if (splitIndex !== -1) {
+      const key: string = lines[i].substring(0, splitIndex).trim();
+      const value: string = lines[i].substring(splitIndex + 1).trim();
+      headers[key] = value;
+    }
+  }
+  if (emptyIndex !== -1) {
+    content = lines.slice(emptyIndex + 1).join('\n').trim();
+  }
+  return { headers, content };
+}
+
 function Details({ item }: { item?: HttpHistoryItem }) {
   if (!item) return <></>;
-  let requestTabItems = [
+  const requestMessage: { headers:Record<string, string>, content:string } = revertRawTextToMessage(item.requestRaw);
+  const requestTabItems = [
     {
-      title: "Request Details",
-      content: (
-        <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
-          {item.requestRaw}
+      title: "Formatted Request Details",
+      content:(
+        <div>
+          <label style={{fontWeight:"bold"}}>Header</label>
+          <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+            {Object.entries(requestMessage.headers).map(([key, value], index) => (
+              <div key={index}>
```

Comment:
since key value are parsed, maybe we can use a list or a table for these headers
https://react.fluentui.dev/?path=/docs/components-datagrid--default